lib/io/storage/StoragePath.js

```diff
@@ -1,6 +1,20 @@
 import crypto from 'crypto';
```

Comment:
this file feels like the kind of thing that really, really, really needs some kind of fuzz testing.  (same with our reports - we should be making database-backed tests for this!)